src/Commands/MeltorCommand.php

```diff
@@ -0,0 +1,309 @@
+     * The console command description.
+     *
+     * @var string
+     */
+    protected $description = 'Create a new migration based on the current MySQL database';
+
+    // Internals
+    protected array $indexes = [];
+    protected array $problems = [];
+    protected array $spatialIndexes = [];
+    protected array $primaryKeyIsSetFor = [];
+    protected bool $canAccessInnoDbIndexes = false;
+    protected bool $ignoreProblems = false;
+    protected bool $warnAboutFloat = false;
+    protected bool $separateForeignKeys = false;
+    protected string $databaseName = '';
+    protected ?ConnectionInterface $dataConnection;
+    protected ?ConnectionInterface $schemaConnection;
+    protected ?Collection $structure;
+    protected ?Meltor $meltor;
+
+    /**
+     * Execute the console command.
+     *
+     * @return int
+     *
+     * @throws Exception
+     */
+    public function handle(): int
+    {
+        $this->meltor              = app('meltor');
+        $this->ignoreProblems      = $this->option('ignoreProblems');
+        $this->separateForeignKeys = $this->option('separateForeignKeys');
+        $this->dataConnection      = DB::connection($this->meltor->config('connection.data'));
+        $this->databaseName        = $this->dataConnection->getDatabaseName();
+        $schemaConnectionName      = $this->meltor->config('connection.schema');
+
+        try {
+            $this->schemaConnection = DB::connection($schemaConnectionName);
+        } catch (Exception $exception) {
+            $this->warn(sprintf('Please configure a database connection named %s that points to the information_schema database. See README.md.', $schemaConnectionName));
+            return 1;
+        }
+
+        if ($this->option('restore')) {
+            $this->meltor->restoreBackup($this);
+
+            return 0;
+        }
+
+        $this->newLine();
+        $this->line(sprintf('Meltor tries to consolidate all past migrations, based on this system\'s "%s" DB', $this->databaseName));
+        $this->line('Make sure all your systems have been deployed, and that this system has your current, production database structure.');
+
+        if ($this->option('testrun')) {
+            $this->newLine();
+            $this->line('Test run enabled');
+            if (!$this->option('force')) {
+                $ready = $this->ask(
+                    'Please confirm that you have deleted all old migrations, except ones that change framework or package created tables (y|N)'
+                );
+
+                if ($ready != 'y') {
+                    $this->info('Aborting');
+
+                    return 0;
+                }
+            }
+        }
+
+        $this->newLine();
+        $this->line(sprintf('Reading structure information from connection %s, database %s', $this->meltor->config('connection.data'), $this->databaseName));
+
+        $this->structure   = $this->meltor->getDatabaseStructure($this->databaseName, $this->schemaConnection);
```

Comment:
Why do you need to pass databaseName and schemaConnection on every method call instead of initializing meltor with that on the constructor?